subsys/net/lib/coap/coap.c

```diff
@@ -1172,7 +1172,7 @@ bool coap_packet_is_request(const struct coap_packet *cpkt)
 {
 	uint8_t code = coap_header_get_code(cpkt);
 
-	return !(code & ~COAP_REQUEST_MASK);
+	return (code != COAP_CODE_EMPTY) && !(code & ~COAP_REQUEST_MASK);
```

Comment:
There is a "ping/pong" definition in RFC 7252 (CON 0.0 => RST 0.0), where, depending on the implementation, an empty CON message may be considered to be a request. 
Not sure, if that is related, but that change may fail with a ping/pong.

Reply:
I was tempted to change this initially, but TBH after digging into RFC 7252 I'm not sure if we should consider such an empty CON "probe" as a request. Specifically, chapter 4.1 defines an empty message separately from request/response:
> Separate from the message type, a message may carry a request, a response, or be Empty.

Similarily in chapter 4.2 and empty message is considered separately from regular request/response:
> A Confirmable message always carries either a request or response, unless it is used only to elicit a Reset message, in which case it is Empty.

Of course if the above assumption is not a valid one I guess we could just add an extra type check to the function instead.

Reply:
The one view is the wording, and with that the change is well.
The other view is the processing of such an CON 0.0.
I'm not sure, what the logic inside the zephyr functions for that ping/pong is. I'm still on NCS 2.6.2 so I don't see the current implementation (I would need to check that out to see, how such a ping pong is intended to be processed. I will go for that tomorrow.)